git_machete/git_operations.py

```diff
@@ -454,9 +454,9 @@ def get_strict_remote_for_fetching_of_branch(self, branch: LocalBranchShortName)
         remote = self.get_config_attr_or_none(f"branch.{branch}.remote")
         return remote.rstrip() if remote else None
 
-    def get_combined_remote_for_fetching_of_branch(self, branch: LocalBranchShortName) -> Optional[str]:
+    def get_combined_remote_for_fetching_of_branch(self, branch: LocalBranchShortName, remotes: Optional[List[str]] = None) -> Optional[str]:
```

Comment:
Hmmm okay so the purpose of this param is to **narrow down** the list of remotes that can be taken into account? 🤔